metadata_fetcher/fetchers/Fetcher.py

```diff
@@ -78,29 +78,34 @@ def fetch_page(self):
             raise FetchError(
                 f"[{self.collection_id}]: unable to fetch page {page}")
 
+
         if self.check_page(response):
+            text = self.get_text_from_response(response)
             if settings.DATA_DEST == 'local':
-                self.fetchtolocal(response.text)
+                self.fetchtolocal(text)
             else:
-                self.fetchtos3(response.text)
+                self.fetchtos3(text)
```

Comment:
Because the metadata for each item is not fetched from the same URL that provides the paginated list of items (by reference) and two additional further HTTP requests are required to fetch the metadata for the page, an additional method was added to allow the generation of a pseudo-response that contains the paginated list with the item metadata. This method `get_text_from_response`, by default, will return `resolve.text`, see below.